web/src/popover_menus.js

```diff
@@ -186,6 +187,12 @@ export function toggle_message_actions_menu(message) {
         return true;
     }
 
+    const $selected_row = message_lists.current.selected_row();
+    // If we do scroll to the top then we should suppress the scroll hide.
+    if (message_viewport.view_message_top($selected_row)) {
+        popovers.set_suppress_scroll_hide();
```

Comment:
Why do we do this here, rather than having `message_viewport.view_message_top` take care of it?

Reply:
Hmm, yeah I think it makes more sense to allow `maybe_scroll_to_show_message_top` to handle those instead. I originally thought the author did this for reusability of the function but now that I think about it, it's fine if we more it over to `maybe_scroll_to_show_message_top` to handle it and it makes more sense to do so.